src/main/java/bookstore/service/user/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package bookstore.service.user.impl;
+
+import bookstore.dto.user.UserRegistrationRequestDto;
+import bookstore.dto.user.UserResponseDto;
+import bookstore.exception.RegistrationException;
+import bookstore.mapper.UserMapper;
+import bookstore.model.User;
+import bookstore.repository.UserRepository;
+import bookstore.service.user.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto request)
+            throws RegistrationException {
```

Comment:
```suggestion
    public UserResponseDto register(UserRegistrationRequestDto request) {
```

Reply:
Then I won't be able to throw an error when registering with an existing email